earthfile2llb/converter.go

```diff
@@ -1535,11 +1535,12 @@ func (c *Converter) prepBuildTarget(ctx context.Context, fullTargetName string,
 		opt.DoSaves = c.opt.DoSaves && !target.IsRemote()   // legacy mode only saves artifacts from local targets
 		opt.DoPushes = c.opt.DoPushes && !target.IsRemote() // legacy mode only saves artifacts from local targets
 	}
+	opt.DoSaves = opt.DoSaves && allowDoSaves
```

Comment:
Wouldn't this already be false on `WITH DOCKER` by the virtue of `WITH DOCKER` calling `BuildAsync` with the `cmdT` set to `LOAD`?

This condition would evaluate to false anyway, in this situation:

```Go
opt.DoSaves = (cmdT == buildCmd && c.opt.DoSaves && !c.opt.OnlyFinalTargetImages)
```

Perhaps the test you were trying to fix was testing the old implementation, where `ReferencedSaveOnly` was `false`? I think we should try not to maintain that combination.

Reply:
> Perhaps the test you were trying to fix was testing the old implementation, where `ReferencedSaveOnly` was `false`? I think we should try not to maintain that combination.

That's likely what happened, i'll try backing this out, and might very well need to expand the overrides to be: `--version-flag-overrides="wait-block,use-copy-include-patterns,referenced-save-only"